docs/user/install-gke-external.md

```diff
@@ -0,0 +1,229 @@
+
+## Introduction
+
+In this guide you will learn how to externally expose the Turbinia application.
+This guide is recommended for users who have already deployed Turbinia to a GKE
+cluster, but would like to access the Web UI and API server through an externally
+available URL instead of port forwarding the Turbinia service locally from the cluster.
+
+### Prerequisites
+
+- A Google Cloud Account and a GKE cluster with Turbinia deployed
+- The ability to create GCP resources
+- `gcloud` and `kubectl` locally installed
+
+## Deployment
+
+Please follow the steps below for configuring Turbinia to be externally accessible.
+
+### 1. Create a static external IP address
+
+- Create a global static IP address as follows:
+
+```
+gcloud compute addresses create turbinia-webapps --global
+```
+
+- You should see the new IP address listed:
+
+```
+gcloud compute addresses list
+```
+
+### 2. Create a domain or use pre-existing one
+
+You will need a domain to host Turbinia on. You can either register a new domain in a registrar
+of your choice or use a pre-existing one.
+
+To do so through `gcloud`, search for a domain that you want to register as follows:
+
+```
+gcloud domains registrations search-domains SEARCH_TERMS
+```
+
+If the domain is available, then you can register the domain through:
+
+```
+gcloud domains registrations register DOMAIN_NAME
+```
+
+Once the domain is registered, you will need to create a DNS `A` record pointing
+to the external IP address created above, which can be done through the external
+provider you registered the domain from or through `gcloud` as shown below.
+
+First you will need to create the managed DNS zone: Enter the following while also
+adjusting the `--dns-name` flag to the domain you registered:
+
+```
+ gcloud dns managed-zones create turbinia-dns --dns-name DOMAIN_NAME --description "Turbinia managed DNS"
+```
+
+Then add the DNS `A` record pointing to the external IP address created above as follows:
+
+```
+gcloud dns record-sets create DOMAIN_NAME --zone="turbinia-dns" --type="A" --ttl="300" --rrdatas="EXTERNAL_IP"
+```
+
+If you would like to register a domain and update its DNS record through Google Domains instead,
+you can skip this step altogether and follow handy instructions provided [here](https://cert-manager.io/docs/tutorials/getting-started-with-cert-manager-on-google-kubernetes-engine-using-lets-encrypt-for-ingress-ssl/#4-create-a-domain-name-for-your-website). Additionally, DNS can be managed through [ExternalDNS](https://github.com/kubernetes-sigs/external-dns), however setup is outside the scope of this documentation.
+
+### 3. Create Oauth2 Application IDs
+
+Authentication is handled by a proxy utility named [Oauth2 Proxy](https://oauth2-proxy.github.io/oauth2-proxy/). We will be configuring the Oauth2 Proxy with Google Oauth, however there are alternative [providers](https://oauth2-proxy.github.io/oauth2-proxy/docs/configuration/oauth_provider) that you may configure instead.
```

Comment:
Maybe s/We will be configuring/This guide will be walking through configuring/?

Reply:
Ack